packages/ember-glimmer/lib/renderer.js

```diff
@@ -36,118 +36,65 @@ class DynamicScope {
   }
 }
 
-class SchedulerRegistrar {
-  constructor() {
-    this.callback = null;
-    this.callbacks = null;
-
-    backburner.on('begin', (arg1, arg2) => {
-      if (this.callback) {
-        this.callback(arg1, arg2);
-      } else if (this.callbacks) {
-        for (let i = 0; i < this.callbacks.length; ++i) {
-          this.callbacks[i](arg1, arg2);
-        }
-      }
-    });
-  }
+const renderers = [];
 
-  register(callback) {
-    if (this.callbacks) {
-      this.callbacks.push(callback);
-    } else if (!this.callback) {
-      this.callback = callback;
-    } else {
-      this.callbacks = [this.callback, callback];
-      this.callback = null;
-    }
-  }
+setHasViews(() => renderers.length > 0);
 
-  deregister(callback) {
-    let foundCallback = false;
-    if (this.callbacks) {
-      let callbackIndex = this.callbacks.indexOf(callback);
-      foundCallback = ~callbackIndex;
-      if (foundCallback) {
-        this.callbacks.splice(callbackIndex, 1);
-      }
-    } else if (this.callback === callback) {
-      this.callback = null;
-      foundCallback = true;
-    }
-
-    if (!foundCallback) {
-      throw new TypeError('Cannot deregister a callback that has not been registered.');
-    }
-  }
-
-  hasRegistrations() {
-    return !!this.callback || !!this.callbacks && !!this.callbacks.length;
-  }
+function register(renderer) {
+  assert('renderer is not registered', renderers.indexOf(renderer) === -1);
+  renderers.push(renderer);
 }
-const schedulerRegistrar = new SchedulerRegistrar();
-
-setHasViews(function rendererHasViews() {
-  return schedulerRegistrar.hasRegistrations();
-});
 
-class Scheduler {
-  constructor() {
-    this._root = null;
-    this._scheduleMaybeUpdate = () => {
-      run.backburner.schedule('render', this, this._maybeUpdate, CURRENT_TAG.value());
-    };
-  }
-
-  destroy() {
-    if (this._root) {
-      this.deregisterView(this._root);
-    }
-  }
+function deregister(renderer) {
+  let index = renderers.indexOf(renderer);
+  assert('renderer is registered', index !== -1);
+  renderers.splice(index, 1);
+}
 
-  registerView(view) {
-    if (!this.root) {
-      schedulerRegistrar.register(this._scheduleMaybeUpdate);
-      this._root = view;
-    } else {
-      throw new TypeError('Cannot register more than one root view.');
-    }
-  }
-
-  deregisterView(view) {
-    if (this._root === view) {
-      this._root = null;
-      schedulerRegistrar.deregister(this._scheduleMaybeUpdate);
-    }
+function loopBegin() {
+  for (let i = 0; i < renderers.length; i++) {
+    renderers[i]._scheduleRevalidate();
   }
+}
 
-  _maybeUpdate(lastTagValue) {
-    if (CURRENT_TAG.validate(lastTagValue)) { return; }
-    let view = this._root;
-    if (view) {
-      view.renderer.rerender(view);
+function K() {}
+
+let loop = 0;
+function loopEnd(current, next) {
+  for (let i = 0; i < renderers.length; i++) {
```

Comment:
reset `loops` at the beginning, in case something went wrong last time and we didn't reach the end?